tests/components/prometheus/test_init.py

```diff
@@ -639,6 +640,351 @@ async def test_lock(hass, hass_client):
     )
 
 
+async def test_renaming_entity_name(hass, hass_client):
+    """Test renaming entity name."""
+    assert await async_setup_component(
+        hass,
+        "conversation",
+        {},
+    )
+    client = await setup_prometheus_client(hass, hass_client, "")
+
+    assert await async_setup_component(
+        hass, sensor.DOMAIN, {"sensor": [{"platform": "demo"}]}
+    )
+
+    await hass.async_block_till_done()
+    body = await generate_latest_metrics(client)
+
+    assert (
+        'sensor_temperature_celsius{domain="sensor",'
+        'entity="sensor.outside_temperature",'
+        'friendly_name="Outside Temperature"} 15.6' in body
+    )
+
+    assert (
+        'entity_available{domain="sensor",'
+        'entity="sensor.outside_temperature",'
+        'friendly_name="Outside Temperature"} 1.0' in body
+    )
+
+    assert (
+        'sensor_humidity_percent{domain="sensor",'
+        'entity="sensor.outside_humidity",'
+        'friendly_name="Outside Humidity"} 54.0' in body
+    )
+
+    assert (
+        'entity_available{domain="sensor",'
+        'entity="sensor.outside_humidity",'
+        'friendly_name="Outside Humidity"} 1.0' in body
+    )
+
+    registry = entity_registry.async_get(hass)
+    assert "sensor.outside_temperature" in registry.entities
+    registry.async_update_entity(
+        entity_id="sensor.outside_temperature",
+        name="Inside Temperature",
+    )
+
+    await hass.async_block_till_done()
+    body = await generate_latest_metrics(client)
+
+    # Check if old metrics deleted
+    assert (
+        'sensor_temperature_celsius{domain="sensor",'
+        'entity="sensor.outside_temperature",'
+        'friendly_name="Outside Temperature"} 15.6' not in body
+    )
+
+    assert (
+        'entity_available{domain="sensor",'
+        'entity="sensor.outside_temperature",'
+        'friendly_name="Outside Temperature"} 1.0' not in body
+    )
+
+    # Check if new metrics created
+    assert (
+        'sensor_temperature_celsius{domain="sensor",'
+        'entity="sensor.outside_temperature",'
+        'friendly_name="Inside Temperature"} 15.6' in body
+    )
+
+    assert (
+        'entity_available{domain="sensor",'
+        'entity="sensor.outside_temperature",'
+        'friendly_name="Inside Temperature"} 1.0' in body
+    )
+
+    # Keep other sensors
+    assert (
+        'sensor_humidity_percent{domain="sensor",'
+        'entity="sensor.outside_humidity",'
+        'friendly_name="Outside Humidity"} 54.0' in body
+    )
+
+    assert (
+        'entity_available{domain="sensor",'
+        'entity="sensor.outside_humidity",'
+        'friendly_name="Outside Humidity"} 1.0' in body
+    )
+
```

Comment:
Again, perhaps this can be simplified with a broader check, like `assert 'friendly_name="Outside Humidity"' in '\n'.join(body)`?

Reply:
This would stop at the first match. If one of the metrics would be deleted by mistake, the test wouldn't notice it.
On the other hand we're not checking every metric for `sensor.outside_humidity` at the moment anyway.
Should I change it regardless?

Reply:
Oh yes, you are right, it's safer to have more specific checks here.